shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.test.js

```diff
@@ -603,3 +610,126 @@ describe('serveCaseToIrsInteractor', () => {
+      applicationContext,
+      caseEntity,
+      user,
+    });
+
+    const addedDocketRecord = caseEntity.docketEntries.find(
+      docketEntry => docketEntry.eventCode === 'FEE',
+    );
+
+    expect(addedDocketRecord).toBeDefined();
+    expect(addedDocketRecord.filingDate).toEqual('Today');
+  });
+
+  it('adds a docketRecord for a waived petition payment', async () => {
+    const caseEntity = new Case(
+      {
+        ...MOCK_CASE,
+        docketEntries: [],
+        petitionPaymentStatus: PAYMENT_STATUS.WAIVED,
+        petitionPaymentWaivedDate: 'Today',
+        petitioners: undefined,
+      },
+      { applicationContext },
+    );
+    await addDocketEntryForPaymentStatus({
+      applicationContext,
+      caseEntity,
+      user,
+    });
+
+    const addedDocketRecord = caseEntity.docketEntries.find(
+      docketEntry => docketEntry.eventCode === 'FEEW',
+    );
+
+    expect(addedDocketRecord).toBeDefined();
+    expect(addedDocketRecord.filingDate).toEqual('Today');
+  });
+});
+
+describe('addDocketEntryForNANE', () => {
+  let user;
+
+  beforeEach(() => {
+    user = applicationContext.getCurrentUser();
+  });
+
+  it('should not increase the docket entries and not upload a generated pdf if noticeOfAttachments is false', async () => {
+    const caseEntity = new Case(
+      {
+        ...MOCK_CASE,
+        noticeOfAttachments: false,
+      },
+      { applicationContext },
+    );
+
+    const docketEntriesFromNewCaseCount = caseEntity.docketEntries.length;
+
+    await addDocketEntryForNANE({
+      applicationContext,
+      caseEntity,
+      user,
+    });
+
+    expect(caseEntity.docketEntries.length).toEqual(
+      docketEntriesFromNewCaseCount,
+    );
+
+    expect(applicationContext.getUtilities().uploadToS3).not.toHaveBeenCalled();
+  });
+
+  it('should increase the docket entries and upload a generated pdf if noticeOfAttachments is true', async () => {
+    const caseEntity = new Case(
+      {
+        ...MOCK_CASE,
+        noticeOfAttachments: true,
+      },
+      { applicationContext },
+    );
+
+    const newDocketEntriesFromNewCaseCount =
+      caseEntity.docketEntries.length + 1;
+
+    await addDocketEntryForNANE({
+      applicationContext,
+      caseEntity,
+      user,
+    });
+
+    expect(caseEntity.docketEntries.length).toEqual(
+      newDocketEntriesFromNewCaseCount,
+    );
+
+    expect(applicationContext.getDocumentGenerators().order).toHaveBeenCalled();
+    const passedInNoticeTitle =
+      applicationContext.getDocumentGenerators().order.mock.calls[0][0].data
+        .orderTitle;
+    expect(passedInNoticeTitle).toEqual(passedInNoticeTitle.toUpperCase()); //asserts that the passed in title was uppercase
+
+    expect(applicationContext.getUtilities().uploadToS3).toHaveBeenCalled();
+    expect(
+      applicationContext.getPersistenceGateway().saveDocumentFromLambda,
+    ).toHaveBeenCalled();
+    // eslint-disable-next-line max-lines
```

Comment:
Nit: It's probably time to split this into two test files

Reply:
I agree.